### DISCUSSION TITLE
Activate Https on Remotely Server Using Kestrel.

### DISCUSSION THREAD

**Discussion by MagoMaverick** [immense/Remotely#531] [Q&A]
Hi,
installed Remotely on my server and after some tests I notice it work without IIS simply putting this configuration in appSettings.Production.json
 "Kestrel": {
    "Endpoints": {
      "Http": {
        "Url": "http://mydomain.it:4545"
      }
    }
  }

It works, but now, to have in production I need to use Https. I already create a certificate using win-acme.
I need only to understand what can be the correct configuration on appSettings. I tried:
"Kestrel": {
    "Endpoints": {
      "Https": {
	"Url": "https://mydomain.it:4545"
      }
    }
  }

but when I try to reach server I get error: ERR_SSL_PROTOCOL_ERROR The site cannot provide a secure connection (using Chrome).

Somebody know how to configure it?

Thanks in advance.

**Comment by goftari** [comment]
Hi
You will have to copy the appsettings.Development.json to appsettings.Production.json and add a new HTTPS endpoint like this for Kestrel:
```
  "Kestrel": {
    "Endpoints": {
      "Http": {
        "Url": "http://0.0.0.0"
      },
      "Https": {
        "Url": "https://{IP ADDRESS which could be 0.0.0.0 without brackets}",
        "Certificate": {
          "Subject": "{certificate name without brackets}",
          "Store": "My",
          "Location": "LocalMachine",
          "AllowInvalid": "{true if using self-signed certificate without brackets}" 
        }
      }
    }
  }
```

You may also find this helpful: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-7.0
